src/sidebar/util/test/annotation-sharing-test.js

```diff
@@ -24,75 +25,129 @@ describe('sidebar.util.annotation-sharing', () => {
     sharingUtil.$imports.$restore();
   });
 
-  describe('#annotationSharingEnabled', () => {
-    it('returns true if no service settings present', () => {
-      fakeServiceConfig.returns(null);
-      assert.isTrue(sharingUtil.sharingEnabled({}));
-    });
+  describe('annotationSharingLink', () => {
+    context('sharing an annotation whose document is on the web', () => {
+      it('returns `incontext` link if set on annotation', () => {
+        assert.equal(
+          sharingUtil.annotationSharingLink(fakeAnnotation),
+          'https://www.example.com'
+        );
+      });
 
-    it('returns true if service settings do not have a `enableShareLinks` prop', () => {
-      // service config is an empty object
-      assert.isTrue(sharingUtil.sharingEnabled({}));
-    });
+      it('returns `html` link if `incontext` link not set on annotation', () => {
+        delete fakeAnnotation.links.incontext;
 
-    it('returns true if service settings `enableShareLinks` is non-boolean', () => {
-      fakeServiceConfig.returns({ enableShareLinks: 'foo' });
-      assert.isTrue(sharingUtil.sharingEnabled({}));
+        assert.equal(
+          sharingUtil.annotationSharingLink(fakeAnnotation),
+          'https://www.example2.com'
+        );
+      });
+
+      it('returns `undefined` if links empty', () => {
+        delete fakeAnnotation.links.incontext;
+        delete fakeAnnotation.links.html;
+
+        assert.isUndefined(sharingUtil.annotationSharingLink(fakeAnnotation));
+      });
+
+      it('returns `undefined` if no links on annotation', () => {
```

Comment:
Can this ever actually happen with the API or will it always set `links` but possibly leave the object empty?

Reply:
I do not know offhand.